ciongke/cmd/test-pr/main.go

```diff
@@ -0,0 +1,207 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"golang.org/x/oauth2"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/kubernetes/test-infra/ciongke/cmdlog"
+	"github.com/kubernetes/test-infra/ciongke/gcs"
+	"github.com/kubernetes/test-infra/ciongke/github"
+	"github.com/kubernetes/test-infra/ciongke/kube"
+)
+
+var (
+	repoURL   = flag.String("repo-url", "", "URL of the repo to test.")
+	repoName  = flag.String("repo-name", "", "Name of the repo to test.")
+	pr        = flag.Int("pr", 0, "Pull request to test.")
+	branch    = flag.String("branch", "", "Target branch.")
+	head      = flag.String("head", "", "Head SHA.")
+	workspace = flag.String("workspace", "/workspace", "Where to checkout the repo.")
+	namespace = flag.String("namespace", "default", "Namespace for all CI objects.")
+	dryRun    = flag.Bool("dry-run", true, "Whether or not to avoid mutating calls to GitHub.")
+
+	sourceBucket = flag.String("source-bucket", "", "Bucket for source tars.")
+
+	githubTokenFile = flag.String("github-token-file", "/etc/oauth/oauth", "Path to the file containing the GitHub OAuth secret.")
+)
+
+type testDescription struct {
+	Name  string `yaml:"name"`
+	Image string `yaml:"image"`
+}
+
+type testClient struct {
+	RepoURL  string
+	RepoName string
+	PRNumber int
+	Branch   string
+	Head     string
+
+	Workspace    string
+	SourceBucket string
+
+	GitHubClient github.Client
+	KubeClient   kube.Client
+	GCSClient    gcs.Client
+	ExecCommand  func(name string, arg ...string) *exec.Cmd
+}
+
+func main() {
+	flag.Parse()
+
+	oauthSecretRaw, err := ioutil.ReadFile(*githubTokenFile)
+	if err != nil {
+		log.Printf("Could not read oauth secret file: %s", err)
+		return
+	}
+	oauthSecret := string(bytes.TrimSpace(oauthSecretRaw))
+
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: oauthSecret})
+	tc := oauth2.NewClient(oauth2.NoContext, ts)
+	var githubClient github.Client
+	if *dryRun {
+		githubClient = github.NewDryRunClient(tc)
+	} else {
+		githubClient = github.NewClient(tc)
+	}
+
+	kubeClient, err := kube.NewClientInCluster(*namespace)
+	if err != nil {
+		log.Printf("Error getting Kubernetes client: %s", err)
+		return
+	}
+
+	gcsClient, err := gcs.NewClient()
+	if err != nil {
+		log.Printf("Error getting GCS client: %s", err)
+		return
+	}
+	client := &testClient{
+		RepoURL:  *repoURL,
+		RepoName: *repoName,
+		PRNumber: *pr,
+		Branch:   *branch,
+		Head:     *head,
+
+		Workspace:    *workspace,
+		SourceBucket: *sourceBucket,
+
+		GitHubClient: githubClient,
```

Comment:
Looking through this, I can see where `KubeClient` will eventually be used, but I don't see anything that needs or will need `GitHubClient`.